src/components/BarList/BarList.tsx

```diff
@@ -0,0 +1,199 @@
+  prop is not provided, it defaults to `'#b5d4fc'`. */
+  defaultBackgroundBarColor?: CSSProperties['color']
+}
+
+/**
+ * This function calculates the widths of bars in a chart based on the values provided.
+ * @param {number[]} values - An array of numbers representing values for which we want to calculate
+ * corresponding widths.
+ * @returns The function `calculateWidthsFromValues` returns an array of numbers representing the
+ * calculated widths based on the input array of numbers `values`. */
+const calculateWidthsFromValues = (values: number[]) => {
+  const MIN_WIDTH = 1
+  const MAX_WIDTH = 100
+
+  const max = Math.max(...values)
+
+  const calculateWidth = (value: number) => {
+    if (value === 0) return 0
+    return Math.max((value / max) * MAX_WIDTH, MIN_WIDTH)
+  }
+
+  const widths = values?.map(calculateWidth)
+
+  return widths
+}
+
+const BarList: FC<BarListProps> = ({
+  classNameBar,
+  fontSizeBar = 'base',
+  heightBar = 'full',
+  listData,
+  roundedBar = 'md',
+  titleMetrics,
+  titleValues,
+  marginYItem = '1',
+  defaultBackgroundBarColor = '#b5d4fc',
+  defaultTextBarColor = '#1D4ED8',
+  ...props
+}) => {
+  const widths = calculateWidthsFromValues(listData.map((item) => item.value))
+
+  return (
+    <div
+      data-testid="container-bar-list"
+      className={composeClasses('w-full h-full', props?.className)}
+      style={props?.style}
+    >
+      {(titleMetrics || titleValues) && (
+        <Flex
+          alignItems="center"
+          justifyContent="between"
+          className="w-full font-medium mb-1"
+        >
+          <p className="text-left">{titleMetrics}</p>
+          <p className="text-right">{titleValues}</p>
+        </Flex>
+      )}
+      {listData?.map((item, index) => {
+        return (
+          <div
+            key={`${item?.label}-${item?.value}`}
+            className={composeClasses('w-full h-full', `my-${marginYItem}`)}
+            data-testid="item-bar"
+          >
+            <Flex
+              alignItems="center"
+              justifyContent="between"
+              className="relative flex-row flex-nowrap w-full h-full"
+            >
+              <Flex
+                alignItems="center"
+                justifyContent="between"
+                className="p-2 flex-row flex-nowrap gap-x-1 z-10 h-full"
+              >
+                {item?.startIcon && item.startIcon}
+                <BarLabel
+                  textBarColor={item.textBarColor || defaultTextBarColor}
+                  fontSizeBar={fontSizeBar}
+                  href={item?.href}
+                >
+                  {item?.label}
+                </BarLabel>
+                {item?.endIcon && item.endIcon}
+              </Flex>
+              <div
+                data-testid="item-background"
+                className={composeClasses(
+                  'absolute',
+                  `h-${heightBar} rounded-${roundedBar}`,
```

Comment:
these validations would not be superfluous, although these props have a default value

```suggestion
                  heightBar && roundedBar && `h-${heightBar} rounded-${roundedBar}`,
```